include/hex.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_HEX_H_
+#define ZEPHYR_INCLUDE_HEX_H_
+
+#include <stddef.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief      Convert a single character into a hexadecimal nibble.
+ *
+ * @param[in]  c     The character to convert
+ * @param      x     The address of storage for the converted number.
+ *
+ *  @param err zero on success or (negative) error code otherwise.
+ */
+int char2hex(char c, u8_t *x);
```

Comment:
Should this and other funcs in this file have some prefix, like sys_ or similar in order to avoid possible naming conflicts

Reply:
I don't know about this. Maybe the code owners for the core libs could answer?